File: 3191. Minimum Operations to Make Binary Array Elements Equal to One I.cpp
class Solution {
    public:
        int minOperations(vector<int>& nums) 
        {
            int ans = 0, n = nums.size(), flag = true;
            for(int i=0;i<n;i++)
            {
                if(not nums[i])
                {
                    int cnt = 0;
                    while (i+cnt < n and cnt < 3)
                    {
                        nums[i+cnt] = (nums[i+cnt] ? 0 : 1);
                        cnt++;
                    }
                    if(cnt == 3) ans++;
                    else flag = false;
                }
            }
            return flag ? ans : -1;
        }
    };

File: 3191. Minimum Operations to Make Binary Array Elements Equal to One I.js
var minOperations = function(nums) 
{
    let ans = 0, n = nums.length;
    for(let i = 0; i < n; i++)
    {
        if(!nums[i] && i + 2 < n)
        {
            ans += 1;
            nums[i] = !nums[i];
            nums[i + 1] = !nums[i + 1];
            nums[i + 2] = !nums[i + 2];
        }
    }

    if(nums[n - 1] && nums[n - 2] && nums[n - 3]) return ans;
    return -1;
};